publish_courses/static/publish_courses/js/controllers/IndexController.js

```diff
@@ -5,164 +5,88 @@
-        // include Ongoing term (year==1900)
+        // fetch active, complete terms for last two years and any future years;
+        // note this does not include Ongoing term (year==1900)
         var currentYear = new Date().getFullYear();
-        var httpGetConfig = {params: {
-            active: 1,
-            calendar_year__gte: currentYear-4,  // last five years
-            limit: 100,
-            ordering: '-end_date,term_code__sort_order',
-            school: $scope.school.id,
-            with_end_date: 'True',
-            with_start_date: 'True',
-        }};
+        var termsGetConfig = { params: {
+                // calendar_year__gte: currentYear - 1,  // last two years
+                calendar_year__gte: currentYear - 2,  // last three years
+                school: $scope.school.id}};
 
-        // todo: move this into shared component
-        $http.get('/icommons_rest_api/api/course/v2/terms/', httpGetConfig)
-            .then(function successCallback(response) {
-                var dropdownSuffix = ' <span class="caret"></span>';
-                $scope.filterOptions.terms =
-                    $scope.filterOptions.terms.concat(
-                        response.data.results.map(function (t) {
-                            return {
-                                key: 'term',
-                                name: t.display_name,  // todo: do we need this?
-                                query: true,
-                                text: t.display_name + dropdownSuffix,  // todo: move into selectTerm()?
-                                value: t.meta_term_id }; }));
-                // todo: implement fetch-until-exhausted
-                if (response.data.next && response.data.next !== '') {
-                    // API returns next=null if there are no more pages
-                    $log.warn('Warning: Some terms missing from dropdown!');
-                }
-                // todo: default to most recent, like in site creator?
-            }, $scope.handleAjaxErrorResponse);
+        atrapi.Terms.getList(termsGetConfig)
+            .then(function gotTerms(terms) { $scope.terms = terms; });
 
+        $scope.clearMessages = function () {
+            $scope.message = null;
+        };
         $scope.loadCoursesSummary = function(){
-            var selectedAccountId= $scope.school.id;
-            var selectedTermId= $scope.filters.terms.value;
-            var config = {params: {
-                    term_id: selectedTermId,
-                    account_id: selectedAccountId
-                }};
-
-            var url = '/publish_courses/api/show_summary';
-            self.dataLoading = true;
-            $http.get(url, config).success(function (data) {
+            $scope.loadingSummary = true;
+            var accountId = $scope.school.id;
+            var termId = $scope.selectedTerm.meta_term_id;
+            pcapi.CourseSummary.get(accountId, termId).then(function (data) {
```

Comment:
As discussed offline, this is handled in the actual service. The ATRAPI, by contrast, exposes it to the consuming controller so it can decide whether/how to use the pending request resolution features.